Update release notes to include all RPC changes since 4.6.0 (#5765)
zcash/zcash: doc/release-notes.md

  `-requirewalletbackup=false` but this is not recommended unless you know what
  you're doing (and have otherwise backed up the wallet's recovery phrase anyway).
  For security reasons, this RPC method is not intended for use via zcash-cli
  but is provided to enable `zcashd-wallet-tool` and other third-party wallet
  interfaces to satisfy the backup confirmation requirement. Use of the
  `walletconfirmbackup` API via zcash-cli would risk that the recovery phrase
  being confirmed might be leaked via the user's shell history or the system
  process table; `zcashd-wallet-tool` is specifically provided to avoid this
  problem.
- `z_getnewaccount` This API allows for creation of new BIP 44 / ZIP 32
  accounts using HD derivation from the wallet's mnemonic seed. Each account represents a
  separate spending authority and source of funds. A single account may contain funds in
  the Sapling and Orchard shielded pools, as well as funds held in transparent addresses.
- `z_getaddressforaccount` This API allows for creation of diversified unified
  addresses under a single account. Each call to this API will, by default, create

>   a new diversified unified address containing p2pkh, Sapling, and Orchard receivers.

Reviewer: ```suggestion
  a new diversified unified address containing P2PKH (transparent), Sapling, and Orchard receivers.
```

  Additional arguments to this API may be provided to request the address to be created
  with a user-specified set of receiver types and diversifier index.
- `z_getbalanceforaccount` This API makes it possible to obtain balance information
  on a per-account basis.

> - `z_getbalanceforviewingkey` This newly created API allows a user to obtain

Reviewer: Inconsistent use of "This API" vs "This newly created API".

  balance information for funds visible to a Sapling or Unified full viewing key; if a
  Sprout viewing key is provided, this method allows retrieval of the balance only in the
  case that the wallet controls the corresponding spending key.  This API has been added
  to supplement (and largely supplant) `z_getbalance`. Querying for balance by a single
  address returns only the amount received by that address, and omits value sent to other
  diversified addresses derived from the same full viewing key; by using
  `z_getbalanceforviewingkey` it is possible to obtain a correct balance that includes all
  amounts controlled by a single spending key, including both those sent to external
  diversified addresses and to wallet-internal change addresses.
- `z_listunifiedreceivers` This API allows the caller to extract the individual
  component receivers from a unified address. This is useful if one needs to

>   provide a bare Sapling or p2pkh address to a service that does not yet

Reviewer: ```suggestion
  provide a bare Sapling or P2PKH address to a service that does not yet
```

  support unified addresses.

RPC Changes
-----------

- The result type for the `listaddresses` endpoint has been modified:
  - The `keypool` source type has been removed; it was reserved but not used.
  - In the `sapling` address results, the `zip32AccountId` attribute has been
    removed in favor of `zip32KeyPath`. This is to allow distinct key paths to
    be reported for addresses derived from the legacy account under different
    child spending authorities, as are produced by `z_getnewaddress`.

> - The results of the 'dumpwallet' and 'z_exportwallet' RPC methods have been modified

Reviewer: ```suggestion
- The results of the `dumpwallet` and `z_exportwallet` RPC methods have been modified
```

  to now include the wallet's newly generated emergency recovery phrase as part of the
  exported data.
- The results of the `getwalletinfo` RPC have been modified to return two new fields:
  `mnemonic_seedfp` and `legacy_seedfp`, the latter of which replaces the field that
  was previously named `seedfp`.
- A new `pool` attribute has been added to each element returned by `z_listunspent`
  to indicate which value pool the unspent note controls funds in.
- `z_listreceivedbyaddress` a `pool` attribute has been added to each result to indicate
  what pool the received funds are held in.
- `z_viewtransaction` has been updated to include attributes that provide
  information about Orchard components of the transaction. Also, the `type`
  attribute for spend and output values has been deprecated and replaced

>   by the `pool` attribute.

Reviewer: I think this would have been clearer to say "a `pool` attribute has been added to the output of the following RPC methods:" and then list them.

- `z_getnotescount` now also returns information for Orchard notes.

> - The output format of `z_exportwallet` has been changed. The exported file now

Reviewer: This sounds scarier than it is. We haven't modified any of the existing non-comment format, but only made changes to data in the comments.

WILL NOT BE RECOVERABLE USING THE EMERGENCY RECOVERY PHRASE UNLESS THEY HAVE
BEEN MOVED TO A NEWLY GENERATED ADDRESS FOLLOWING THE 4.7.0 UPGRADE.

Wallet Updates
--------------

The zcashd wallet has been modified to alter the way that change is handled. In the case
that funds are being spent from a unified account, change is sent to a wallet-internal

> change address for an account instead of sending change amounts back to the original

Reviewer 2: ```suggestion
change address for that account instead of sending change amounts back to the original
```

  corresponds to the `AllowFullyTransparent` policy).

- Since Sprout outputs are no longer created (with the exception of change)
  `z_sendmany` no longer generates payment disclosures (which were only
  available for Sprout outputs) when the `-paymentdisclosure` experimental
  feature flag is set.

RPC Deprecations
----------------

- `z_getnewaddress` has been deprecated in favor of `z_getnewaccount` and
  `z_getaddressforaccount`.
- `z_listaddresses` has been deprecated. Use `listaddresses` instead.
- `z_getbalance` has been deprecated. Use `z_getbalanceforviewingkey` instead.
  See the discussion of how change is now handled under the `Wallet` heading for
  additional background.
- `z_gettotalbalance` has been deprecated. Use `z_getbalanceforaccount` instead.

> - `dumpwallet` has been deprecated. Use `z_exportwallet` instead.

Reviewer: Lots of word duplication here.

Platform Support
----------------

> - Debian 9 has been removed from the list of supported platforms.

Reviewer: No rationale provided here (unlike when we removed Debian 8).
